bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+            then
+                    echo "Unable to sign certificate after $counter seconds"
+                    exit 1
+            fi
+          done
+
+          kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' | base64 -d | openssl x509 -outform pem -text -out ipsec-cert.pem
+
+          kubectl delete csr/$(hostname)
+
+          cat /signer-ca/tls.crt | openssl x509 -outform pem -text > /etc/openvswitch/keys/ipsec-cacert.pem
+
+          ovs-vsctl --retry -t 60 set Open_vSwitch . other_config:certificate=/etc/openvswitch/keys/ipsec-cert.pem \
+                                                     other_config:private_key=/etc/openvswitch/keys/ipsec-privkey.pem \
+                                                     other_config:ca_cert=/etc/openvswitch/keys/ipsec-cacert.pem
+
+          popd
+
+          # Workaround for https://github.com/libreswan/libreswan/issues/373
+          ulimit -n 1024
+
+          /usr/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig
+          # Check for kernel modules
+          /usr/libexec/ipsec/_stackmanager start
+          # Check for nss database status and migration
+          /usr/sbin/ipsec --checknss
+          # Check for nflog setup
+          /usr/sbin/ipsec --checknflog
+          # Start the actual IKE daemon
+          /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --logfile /var/log/openvswitch/libreswan.log
+
+          # Workaround for https://mail.openvswitch.org/pipermail/ovs-dev/2020-October/375734.html
+          OVS_LOGDIR=/var/log/openvswitch OVS_RUNDIR=/var/run/openvswitch OVS_PKGDATADIR=/usr/share/openvswitch /usr/share/openvswitch/scripts/ovs-ctl --ike-daemon=libreswan start-ovs-ipsec
+
+          # Wait for the nbdb container in the ovnkube-master pod to become ready before we try to write to
+          # it.
+          if kubectl wait --for=condition=Ready pod -l app=ovnkube-master -n ${ovn_kubernetes_namespace} --timeout 600s; then
+            echo "ovnkube-master is not Ready, unable to write to OVNNB DB"
+            exit 2
+          else
+            ovn-nbctl set nb_global . ipsec=true
+          fi
+
+          while true; do
+            sleep 10
+          done
+        env:
+        - name: OVS_LOG_LEVEL
+          value: info
+        - name: K8S_NODE
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - mountPath: /var/run/openvswitch
+          name: host-var-run-ovs
+        - mountPath: /var/log/openvswitch/
+          name: host-var-log-ovs
+        - mountPath: /var/run/ovn/
+          name: host-var-run-ovn
+        - mountPath: /signer-ca
+          name: signer-ca
+        - mountPath: /etc/openvswitch
+          name: etc-openvswitch
+        resources:
+        # TODO what is reasonable here?
+          requests:
+            cpu: 100m
+            memory: 300Mi
+          limits:
```

Comment:
I don't know what is reasonable, but usually it's better to not set `limits` as that might cause the pod to be CPU/Mem throttled if it exceeds these limits, leading to bugs which are difficult to pin-point.

Reply:
Yeah, I will do that then. I wasnt sure if there was a convention here?

Reply:
I don't know of any official convention, but I would say that limits serve a bigger purpose for user deployed applications for which the CPU and Memory consumption has been studied in advance. In those cases, a user could define limits on pods as to avoid crashing a node if there's a memory leak in the component. This is however a system critical component, and we shouldn't be restricting it, because if the CPU / Mem spikes at a given moment (maybe because of too much traffic I guess?) then the pod gets throttled (i.e: it gets less access to the CPU) by the scheduler.